src/EnergyPlus/WindowManager.cc

```diff
@@ -540,46 +561,46 @@ namespace WindowManager {
+                        wm->tmpReflectVisBeamBack = visibleSpectrumAverage(state, rbb[0]);
 
                         // set this material to average spectral data
                         thisMaterial->GlassSpectralDataPtr = 0;
-                        thisMaterial->Trans = state.dataWindowManager->tmpTrans;
-                        thisMaterial->TransVis = state.dataWindowManager->tmpTransVis;
-                        thisMaterial->ReflectSolBeamFront = state.dataWindowManager->tmpReflectSolBeamFront;
-                        thisMaterial->ReflectSolBeamBack = state.dataWindowManager->tmpReflectSolBeamBack;
-                        thisMaterial->ReflectVisBeamFront = state.dataWindowManager->tmpReflectVisBeamFront;
-                        thisMaterial->ReflectVisBeamBack = state.dataWindowManager->tmpReflectVisBeamBack;
+                        thisMaterial->Trans = wm->tmpTrans;
+                        thisMaterial->TransVis = wm->tmpTransVis;
+                        thisMaterial->ReflectSolBeamFront = wm->tmpReflectSolBeamFront;
+                        thisMaterial->ReflectSolBeamBack = wm->tmpReflectSolBeamBack;
+                        thisMaterial->ReflectVisBeamFront = wm->tmpReflectVisBeamFront;
+                        thisMaterial->ReflectVisBeamBack = wm->tmpReflectVisBeamBack;
                         SpecDataNum = 0;
                     }
                 }
 
                 if (SpecDataNum == 0 && !thisMaterial->GlassSpectralAndAngle) { // No spectral data for this layer; use spectral average values
                     lquasi = true;
-                    state.dataWindowManager->numpt[IGlass - 1] = 2;
-                    state.dataWindowManager->t[IGlass - 1][0] = thisMaterial->Trans;
+                    numpt[IGlass - 1] = 2;
+                    t[IGlass - 1][0] = thisMaterial->Trans;
                     if (IGlass == 1 || (IGlass == 2 && StormWinConst))
-                        state.dataWindowManager->t[IGlass - 1][0] *= thisMaterial->GlassTransDirtFactor;
-                    state.dataWindowManager->t[IGlass - 1][1] = thisMaterial->TransVis;
+                        t[IGlass - 1][0] *= thisMaterial->GlassTransDirtFactor;
+                    t[IGlass - 1][1] = thisMaterial->TransVis;
                     if (IGlass == 1 || (IGlass == 2 && StormWinConst))
-                        state.dataWindowManager->t[IGlass - 1][1] *= thisMaterial->GlassTransDirtFactor;
-                    state.dataWindowManager->rff[IGlass - 1][0] = thisMaterial->ReflectSolBeamFront;
-                    state.dataWindowManager->rbb[IGlass - 1][0] = thisMaterial->ReflectSolBeamBack;
-                    state.dataWindowManager->rff[IGlass - 1][1] = thisMaterial->ReflectVisBeamFront;
-                    state.dataWindowManager->rbb[IGlass - 1][1] = thisMaterial->ReflectVisBeamBack;
+                        t[IGlass - 1][1] *= thisMaterial->GlassTransDirtFactor;
```

Comment:
None of these things needed to be state variables. They are all local. And because they are `std::array` and not `std::vector` there is no construction/destruction overhead per call.